boards/PDM/Src/Io/Io_Efuse.c

```diff
@@ -0,0 +1,821 @@
+
+    // Read original content of GCR Register
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2ReadRegister(SI_GCR_ADDR, reg_val, e_fuse));
+
+    CLEAR_BIT(reg_val[0], (CSNS1_EN_MASK | CSNS0_EN_MASK));
+    SET_BIT(reg_val[0], (selection & (CSNS1_EN_MASK | CSNS0_EN_MASK)));
+
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, reg_val[0], e_fuse));
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode Io_Efuse_Aux1Aux2ConfigureEfuse(struct Efuse *e_fuse)
+{
+    exit_ok_or_return(Io_Efuse_Aux1Aux2ExitFailSafeMode(e_fuse));
+
+    // Global config register
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, GCR_CONFIG, e_fuse));
+
+    // Channel 0 config registers
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_RETRY_0_ADDR, RETRY_CONFIG, e_fuse));
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_CONFR_0_ADDR, CONFR_CONFIG, e_fuse));
+    exit_ok_or_return(Io_Efuse_Aux1Aux2WriteRegister(
+        SI_OCR_0_ADDR, OCR_LOW_CURRENT_SENSE_CONFIG, e_fuse));
+
+    // Channel 1 config registers
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_RETRY_1_ADDR, RETRY_CONFIG, e_fuse));
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_CONFR_1_ADDR, CONFR_CONFIG, e_fuse));
+    exit_ok_or_return(Io_Efuse_Aux1Aux2WriteRegister(
+        SI_OCR_1_ADDR, OCR_LOW_CURRENT_SENSE_CONFIG, e_fuse));
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ExitFailSafeMode(struct Efuse *e_fuse)
+{
+    // Set WDIN bit for next write
+    // 1_1_00000_00000_0000
+    e_fuse->wdin_bit_to_set = true;
+
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_STATR_0_ADDR, 0x0000, e_fuse));
+    // Disable watchdog
+    exit_ok_or_return(
+        Io_Efuse_Aux1Aux2WriteRegister(SI_GCR_ADDR, GCR_CONFIG, e_fuse));
+
+    // Check if the the efuse is still in fail-safe mode
+    if (HAL_GPIO_ReadPin(FSOB_AUX1_AUX2_GPIO_Port, FSOB_AUX1_AUX2_Pin) ==
+        GPIO_PIN_RESET)
+    {
+        return EXIT_CODE_UNIMPLEMENTED;
+    }
+
+    return EXIT_CODE_OK;
+}
+
+static ExitCode Io_Efuse_Aux1Aux2WriteRegister(
+    uint8_t       register_address,
+    uint16_t      register_value,
+    struct Efuse *e_fuse)
+{
+    return Io_Efuse_WriteRegister(
+        register_address, register_value, CSB_AUX1_AUX2_GPIO_Port,
+        CSB_AUX1_AUX2_Pin, e_fuse);
+}
+
+static ExitCode Io_Efuse_Aux1Aux2ReadRegister(
+    uint8_t       register_address,
+    uint16_t *    register_value,
+    struct Efuse *e_fuse)
+{
+    return Io_Efuse_ReadRegister(
+        register_address, register_value, CSB_AUX1_AUX2_GPIO_Port,
+        CSB_AUX1_AUX2_Pin, e_fuse);
+}
+
+static ExitCode Io_Efuse_WriteRegister(
+    uint8_t       register_address,
+    uint16_t      register_value,
+    GPIO_TypeDef *ChipSelect_GPIO_Port,
+    uint16_t      ChipSelect_GPIO_Pin,
+    struct Efuse *e_fuse)
+{
+    ExitCode exit_code;
+    uint16_t command = 0x0000U;
```

Comment:
Command value of 0x0000 is currently unused

Reply:
Applies to Io_Efuse_ReadRegister as well

Reply:
_0x0000_ is the starting value before setting the bits we want? If I just put `uint16_t command;` instead the compiler _should_ initialize it to _0x0000_ as well but it's good practice and more explicit to initialize the variable yourself instead of relying on the compiler. Unless you were taking about something?🤔

Reply:
Aren't we initializing it to `(uint16_t)((register_address & EFUSE_ADDR_MASK) << EFUSE_ADDR_SHIFT);` in the line below so setting it to 0 before isn't necessary

Reply:
Added bitwise _or_ operation
https://github.com/UBCFormulaElectric/Consolidated-Firmware/blob/7d24cd5253afdabe70e31efd47bfdc52fb930384/boards/PDM/Src/Io/Io_Efuse.c#L792